CMakeLists.txt

```diff
@@ -66,9 +66,11 @@ if (USE_CPU_EXTENSIONS)
         source_group("Source Files\\intel\\visualc" FILES ${AWS_ARCH_SRC})
 
     elseif(AWS_ARCH_INTEL AND AWS_HAVE_GCC_INLINE_ASM)
```

Comment:
is the inline check correct now? this pulled the inline asm and we really want to check that the intrinsics are available. IIRC there's instances where the assembler installed in the path is newer than the gcc version, would that drift slip through here?

Reply:
Work as expected. We tested on CentOS 8 using gcc 13.1 as 2.36 and Ubuntu 23.04 using gcc 12.2 as 2.4

-- Performing Test AWS_ARCH_INTEL - Success

-- Performing Test AWS_ARCH_ARM64

-- Performing Test AWS_ARCH_ARM64 - Failed

-- Performing Test AWS_ARCH_ARM32

-- Performing Test AWS_ARCH_ARM32 - Failed

-- Performing Test AWS_HAVE_GCC_INLINE_ASM

-- Performing Test AWS_HAVE_GCC_INLINE_ASM - Success

Regarding you question the answer yes, but we can add some checks for intrinsics.

        check_c_source_runs(“

            #include <nmmintrin.h>

            int main() {

                __m128i a = _mm_setzero_si128();

                return 0;

            }

        “ SSE42_SUPPORTED)

        check_c_source_runs(“

            #include <immintrin.h>

            int main() {

                __m512 a = _mm512_setzero_ps();

                return 0;

            }

        “ AVX512_SUPPORTED)

If this doesn't answer your question, please let me know.

Reply:
what about this one, would it help?

https://github.com/awslabs/aws-c-common/blob/87a1b565caf8a0cbb560c475f8529a3632263afd/cmake/AwsSIMD.cmake#L38

Reply:
No, not at all. We will need support for SSE42 and AVX512 intrinsics since AVX2 only supports 256-bit wide vector registers. Should we add sse42 and avx512 intrinsic support to that cmake file or directly to aws checksum CMakeLists.txt?

Reply:
we need to add it to aws-c-common so it can be used more broadly, and so we don't wind up with these scatter shot everywhere. Unfortunately that means we need to coordinate merges now. I'll go ahead and file the PR for aws-c-common and post it here. In the meantime we have some build-chain wiring to sort out anyways and we can get going on that.

For future readers of this, I suspect this might cause some Rust build issues as I doubt AVX512 is turned on by default in cargo and those build flags are usually global to the build process. So if your linker is complaining about it,  try making sure AVX512 is turned on in cargo.

Reply:
https://github.com/awslabs/aws-c-common/pull/1041

Reply:
https://github.com/awslabs/aws-c-common/pull/1041

Reply:
https://github.com/awslabs/aws-c-common/pull/1041

Reply:
https://github.com/awslabs/aws-c-common/pull/1041

Reply:
Re rust build questions: AVX512 should only be a codegen-time thing, I wouldn't expect it to depend on anything configured on the linker.